daemon/daemon_unix.go

```diff
@@ -737,11 +737,13 @@ func verifyDaemonSettings(conf *config.Config) error {
 	if conf.BridgeConfig.Iface != "" && conf.BridgeConfig.IP != "" {
 		return fmt.Errorf("You specified -b & --bip, mutually exclusive options. Please specify only one")
 	}
-	if !conf.BridgeConfig.EnableIPTables && !conf.BridgeConfig.InterContainerCommunication {
-		return fmt.Errorf("You specified --iptables=false with --icc=false. ICC=false uses iptables to function. Please set --icc or --iptables to true")
-	}
-	if conf.BridgeConfig.EnableIP6Tables && !conf.Experimental {
-		return fmt.Errorf("ip6tables rules are only available if experimental features are enabled")
+	if !conf.BridgeConfig.InterContainerCommunication {
+		if !conf.BridgeConfig.EnableIPTables {
+			return fmt.Errorf("You specified --iptables=false with --icc=false. ICC=false uses iptables to function. Please set --icc or --iptables to true")
+		}
+		if !conf.BridgeConfig.EnableIP6Tables {
```

Comment:
If you disable IPv6 at the kernel level and explicitly set `EnableIP6Tables=false` and `InterContainerCommunication=false`, then this check will fail. I think we should make this an error only if IPv6 has been enabled on the default bridge (ie. when `EnableIPv6` is true).

FWIW, we don't barf out if a custom network is created with icc=false, ipv6=true and ip6tables disabled. We should probably make config validation more consistent.

Reply:
Done.

(Is it possible to set `icc=false` for a custom network?)

Reply:
> (Is it possible to set icc=false for a custom network?)

Yeah, you can create one with: `docker network create --opt=com.docker.network.bridge.enable_icc=false`